### PR TITLE
Update node-configuration.asciidoc

### PR DESCRIPTION
**Pull request by lhirlimann**
Add the restricted settings

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/elasticsearch/node-configuration.asciidoc [hunk 1]
```diff
@@ -10,6 +10,8 @@ endif::[]
 
 Any setting defined in the `elasticsearch.yml` configuration file can also be defined for a set of Elasticsearch nodes in the `spec.nodeSets[?].config` section.
 
+Some settings cannot be changed and should not be changed see https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-reserved-settings.html
```
Comment (thbkrkr) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/node-configuration.asciidoc:13]:
```suggestion
Some settings cannot be changed and should not be changed, check <<{p}-reserved-settings>>.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/node-configuration.asciidoc [hunk 2]
```diff
@@ -10,6 +10,8 @@ endif::[]
 
 Any setting defined in the `elasticsearch.yml` configuration file can also be defined for a set of Elasticsearch nodes in the `spec.nodeSets[?].config` section.
 
+Some settings cannot be changed and should not be changed, check <<{p}-reserved-settings>>.
```
Comment (barkbay) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/node-configuration.asciidoc:13]: nit: I don't think we reject a manifest in which settings managed by the operator are changed. So, technically speaking, they can be changed.

```suggestion
Some settings are managed by ECK, it is not recommended to change them, refer to <<{p}-reserved-settings>> for more details.
```

Comment (barkbay) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/node-configuration.asciidoc:13]: I'm also wondering if we should not prefix this with `NOTE:`

**Review by barkbay — APPROVED** [review]
LGTM

**Pull request merged**